src/sage/homology/free_resolution.py

```diff
@@ -858,14 +841,17 @@ def _maps(self):
             sage: from sage.homology.free_resolution import FreeResolution
             sage: S.<x,y,z,w> = PolynomialRing(QQ)
             sage: I = S.ideal([y*w - z^2, -x*w + y*z, x*z - y^2])
-            sage: r = FreeResolution(I)
-            sage: r._maps
+            sage: r = FreeResolution(I)                                                 # optional - sage.libs.singular
+            sage: r._maps                                                               # optional - sage.libs.singular
             [
                                              [-y  x]
                                              [ z -y]
             [z^2 - y*w y*z - x*w y^2 - x*z], [-w  z]
             ]
         """
+        from sage.libs.singular.singular import si2sa_resolution
+        from sage.libs.singular.function import singular_function
+
         # This ensures the first component of the Singular resolution to be a
         # module, like the later components. This is important when the
         # components are converted to Sage modules.
```

Comment:
This file may have been tagged with `sage.libs.singular`.

It seems to me that it's better to tag the file instead of most of examples whenever possible since it is easy to remove the tag to the file (perhaps when we change the dependency hierarchy...) Zero possibility?

Reply:
I agree with you and will change it to a file annotation.

Reply:
Thanks.

Dependency on `sage.libs.singular` is an implementation detail of the free resolution module.  In this case, file annotation seems better, as we could later remove the tag entirely.